src/index.js

```diff
@@ -51,9 +51,9 @@ export default class FacebookTokenStrategy extends OAuth2Strategy {
    * @param {Object} options
    */
   authenticate(req, options) {
-    let accessToken = (req.body && req.body[this._accessTokenField]) || (req.query && req.query[this._accessTokenField]) || (req.headers && req.headers[this._accessTokenField]);
+    let accessToken =  (FacebookTokenStrategy.parseAccessToken(req, this._accessTokenField, this._authorizationField));
```

Comment:
You can implement it more even simpler. Kind of:

```js
let accessToken = FacebookTokenStrategy.lookup(req, this._accessTokenField);

lookup = (req, field) {
  return (
    req.body && req.body[field] ||
    req.query && req.query[field] ||
    req.headers && req.headers[field]
  );
}
```

Then you don't need to implement methods for parse `accessToken` or `refreshToken` and just call method

```js
let accessToken = FacebookTokenStrategy.lookup(req, this._accessTokenField)
let refreshToken = FacebookTokenStrategy.lookup(req, this._refreshTokenField);
```